File: src/gestorAplicacion/operacional/Venta.java
package gestorAplicacion.operacional;


import gestorAplicacion.organizacional.Empleado;


import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.ArrayList;

public class Venta implements Serializable {  
		
		//-----------SERIALIZADOR-----------
		private static final long serialVersionUID = 1L;
		
		private static ArrayList<Venta> ventas;
		static {
			ventas = new ArrayList<Venta>();
		}
		

		public static ArrayList<Venta> getVentas() {
				return ventas;
			}

			public static void setVentas(ArrayList<Venta> ventas) {
				Venta.ventas = ventas;
			}
		
	
		

	//-----------ATRIBUTOS-----------
	private Producto productoVendido;
	private int idVenta;
	private int cantidadVendida;
	private Empleado empleadoComision;
	private LocalDateTime fechaVenta;
	private double comision;
	private Inventario inventario;
	public final static double porcentajeComision = 0.2;
	public static int numVenta;
	
	
	//-----------CONSTRUCTOR-----------
	public Venta(Producto productoVendido, Empleado empleadoComision, LocalDateTime fechaVenta, int cantidadVendida, Inventario inventario) {
		this.productoVendido = productoVendido;
		this.empleadoComision = empleadoComision;
		this.empleadoComision.setProductosVendidos(productoVendido);
		this.fechaVenta = fechaVenta;
		this.cantidadVendida = cantidadVendida;
		Venta.numVenta++;
		this.idVenta = Venta.numVenta;
		this.inventario = inventario;
		this.inventario.actualizarExistencias(productoVendido, cantidadVendida);
		
		ventas.add(this); //del serializador
	}
	
	

	//-----------GETTERS y SETTERS-----------
	public Producto getProductoVendido() {
		return productoVendido;
	}

	public void setProductoVendido(Producto productoVendido) {
		this.productoVendido = productoVendido;
	}

	public Empleado getEmpleadoComision() {
		return empleadoComision;
	}

	public void setEmpleadoComision(Empleado empleadoComision) {
		this.empleadoComision = empleadoComision;
	}

	public LocalDateTime getFechaVenta() {
		return fechaVenta;
	}

	public void setFechaVenta(LocalDateTime fechaVenta) {
		this.fechaVenta = fechaVenta;
	}
	
	public int getCantidadVendida() {
		return this.cantidadVendida;
	}
	
	//-----------OTROS METODOS-----------
	public String toString() {
		return "El empleado "+ empleadoComision.getNombre() +" "+empleadoComision.getApellido()+" "+""
				+ " realiza una venta de "+ productoVendido.getNombreProducto() + " en la fecha "+ fechaVenta+" y recibe una comisión de: " 
				+ calcularComision(productoVendido);
	}
	public double calcularComision(Producto productoVendido) {
		comision= this.productoVendido.getPrecioVenta() * porcentajeComision;
		
		return comision;
	}    
}

File: Python/src/gestorAplicacion/operacional/Venta.py
from gestorAplicacion.organizacional import Empleado



class Venta:

    #-----------ATRIBUTOS-----------
    PORCENTAJECOMISION = 0.2
    numVenta = 0
    _venta=[]         #serializador

    #-----------CONSTRUCTOR-----------
    def __init__(self, productoVendido, empleadoComision, fechaVenta, cantidadVendida, inventario):
        #instance fields found by Java to Python Converter:
        self._productoVendido = productoVendido
        self._empleadoComision = empleadoComision
        self._empleadoComision.setProductosVendidos(productoVendido)
        self._fechaVenta = fechaVenta
        self._cantidadVendida = cantidadVendida
        Venta.numVenta += 1
        self._idVenta = Venta.numVenta
        self._inventario = inventario
        self._inventario.actualizarExistencias(productoVendido, cantidadVendida)
        Venta._venta.append(self)           #serializador




    #-----------GETTERS y SETTERS-----------
    @classmethod           #serializador
    def getVentas(cls):
        return cls._venta

    @classmethod               #serializador
    def setVentas(cls, ventas):
        cls._venta = ventas

    def getProductoVendido(self):
        return self._productoVendido

    def setProductoVendido(self, productoVendido):
        self._productoVendido = productoVendido

    def getEmpleadoComision(self):
        return self._empleadoComision

    def setEmpleadoComision(self, empleadoComision):
        self._empleadoComision = empleadoComision

    def getFechaVenta(self):
        return self._fechaVenta

    def setFechaVenta(self, fechaVenta):
        self._fechaVenta = fechaVenta

    def getCantidadVendida(self):
        return self._cantidadVendida

    #-----------OTROS METODOS-----------
    def __str__(self):
        return "El empleado "+ self._empleadoComision.getNombre() +" "+self._empleadoComision.getApellido()+" "+"" + " realiza una venta de "+ self._productoVendido.getNombreProducto() + " en la fecha "+ str(self._fechaVenta) +" y recibe una comision de: " + str(self.calcularComision(self._productoVendido))

    def calcularComision(self, productoVendido):
        comision= productoVendido.getPrecioVenta() * Venta.PORCENTAJECOMISION

        return comision
